std/math.d

```diff
@@ -6129,52 +6178,28 @@ int feqrel(X)(const X x, const X y) @trusted pure nothrow @nogc
         // always 1 lower than we want, except that if bitsdiff==0,
         // they could have 0 or 1 bits in common.
 
-        static if (F.realFormat == RealFormat.ieeeExtended
-                || F.realFormat == RealFormat.ieeeQuadruple)
-        {
-            int bitsdiff = ( ((pa[F.EXPPOS_SHORT] & F.EXPMASK)
-                              + (pb[F.EXPPOS_SHORT] & F.EXPMASK) - 1) >> 1)
-                              - pd[F.EXPPOS_SHORT];
-        }
-        else static if (F.realFormat == RealFormat.ieeeDouble)
-        {
-            int bitsdiff = (( ((pa[F.EXPPOS_SHORT]&0x7FF0)
-                               + (pb[F.EXPPOS_SHORT]&0x7FF0)-0x10)>>1)
-                               - (pd[F.EXPPOS_SHORT]&0x7FF0))>>4;
-        }
-        else static if (F.realFormat == RealFormat.ieeeSingle)
-        {
-            int bitsdiff = (( ((pa[F.EXPPOS_SHORT]&0x7F80)
-                               + (pb[F.EXPPOS_SHORT]&0x7F80)-0x80)>>1)
-                               - (pd[F.EXPPOS_SHORT]&0x7F80))>>7;
-        }
+        int bitsdiff = (((  (pa[F.EXPPOS_SHORT] & F.EXPMASK)
+                          + (pb[F.EXPPOS_SHORT] & F.EXPMASK)
+                          - (1 << F.EXPSHIFT)) >> 1)
+                        - (pd[F.EXPPOS_SHORT] & F.EXPMASK)) >> F.EXPSHIFT;
         if ( (pd[F.EXPPOS_SHORT] & F.EXPMASK) == 0)
         {   // Difference is subnormal
             // For subnormals, we need to add the number of zeros that
             // lie at the start of diff's significand.
             // We do this by multiplying by 2^^real.mant_dig
             diff *= F.RECIP_EPSILON;
-            return bitsdiff + X.mant_dig - pd[F.EXPPOS_SHORT];
+            return bitsdiff + X.mant_dig - ((pd[F.EXPPOS_SHORT] & F.EXPMASK) >> F.EXPSHIFT);
```

Comment:
It's in the commit msg: https://github.com/kinke/phobos/commit/5e8ee1290404e6b1fc5084fc77959a353e71db7a